azure-quantum/azure/quantum/job/job.py

```diff
@@ -142,21 +142,212 @@ def get_results(self, timeout_secs: float = DEFAULT_TIMEOUT):
+
+        :param timeout_secs: Timeout in seconds, defaults to 300
+        :type timeout_secs: float
+        :return: Results dictionary with histogram shots, or raw results if not a json object.
+        :rtype: typing.Any
+        """
+        if self.results is not None:
+            return self.results
+
+        if not self.has_completed():
+            self.wait_until_completed(timeout_secs=timeout_secs)
+
+        if not self.details.status == "Succeeded":
+            if self.details.status == "Failed" and self._allow_failure_results():
+                job_blob_properties = self.download_blob_properties(self.details.output_data_uri)
+                if job_blob_properties.size > 0:
+                    job_failure_data = self.download_data(self.details.output_data_uri)
+                    raise JobFailedWithResultsError("An error occurred during job execution.", job_failure_data)
+
+            raise RuntimeError(
+                f'{"Cannot retrieve results as job execution failed"}'
+                + f"(status: {self.details.status}."
+                + f"error: {self.details.error_data})"
+            )
+
+        payload = self.download_data(self.details.output_data_uri)
+        try:
+            payload = payload.decode("utf8")
+            results = json.loads(payload)
+
+            if self.details.output_data_format == "microsoft.quantum-results.v2":
+                if "DataFormat" not in results or results["DataFormat"] != "microsoft.quantum-results.v2":
+                    raise ValueError(f"\"DataFormat\" was expected to be \"microsoft.quantum-results.v2\" in the Job results for \"{self.details.output_data_format}\" output format.")
+                if "Results" not in results:
+                    raise ValueError(f"\"Results\" field was expected to be in the Job results for \"{self.details.output_data_format}\" output format.")
+                
+                if len(results["Results"]) < 1:
+                    raise ValueError("\"Results\" array was expected to contain at least one item")
+                
+                results = results["Results"]
+
+                if len(results) == 1: 
+                    results = results[0]
+                    if "Histogram" not in results:
+                        raise ValueError(f"\"Histogram\" array was expected to be in the Job results for \"{self.details.output_data_format}\" output format.")
+
+                    histogram_values = results["Histogram"]
+                    outcome_keys = self._process_outcome(histogram_values)
+
+                    # Re-mapping object {'Histogram': [{"Outcome": [0], "Display": '[0]', "Count": 500}, {"Outcome": [1], "Display": '[1]', "Count": 500}]} to {'[0]': {"Outcome": [0], "Count": 500}, '[1]': {"Outcome": [1], "Count": 500}}
+                    return {hist_val["Display"]: {"outcome": outcome, "count": hist_val["Count"]} for outcome, hist_val in zip(outcome_keys, histogram_values)}
```

Comment:
What would be the recommended way to get the total results count?

Reply:
parse the histogram and sum over the counts... or call get shots and check len of list